File: ScoreKeeperMVVMStage1AddingDatabindinglibrary/app/src/main/java/in/nareshtech/scorekeepermvvm_stage1_addingdatabindinglibrary/MainActivity.java
package in.nareshtech.scorekeepermvvm_stage1_addingdatabindinglibrary;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;
import android.view.View;

import in.nareshtech.scorekeepermvvm_stage1_addingdatabindinglibrary.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    ActivityMainBinding binding;
    private int count = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        /*setContentView(R.layout.activity_main);*/
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main);

        binding.plusBtn.setOnClickListener(this);
        binding.minusBtn.setOnClickListener(this);
    }

    // When any of the plus and minus buttons are clicked, we get to this method
    @Override
    public void onClick(View v) {
        if(v.getId() == R.id.plus_btn)
            count++;
        else
            count--;
        binding.textView.setText(String.valueOf(count));
    }
}

File: Kotlin/app/src/main/java/in/nareshtech/kotlin/MainActivity.kt
package `in`.nareshtech.kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    var count = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tv: TextView = findViewById(R.id.textView2)
        val b : Button = findViewById(R.id.button);

        b.setOnClickListener(View.OnClickListener {
            count++
            tv.text = ""+count
        })
    }
}